econml/drlearner.py

```diff
@@ -0,0 +1,633 @@
+                return self
+
+            def predict(self, Y, T, X=None, W=None, *, sample_weight=None):
+                XW = self._combine(X, W)
+                propensities = self._model_propensity.predict_proba(XW)
+                Y_pred = np.zeros((T.shape[0], T.shape[1] + 1))
+                T_counter = np.zeros(T.shape)
+                Y_pred[:, 0] = self._model_regression.predict(np.hstack([XW, T_counter]))
+                Y_pred[:, 0] += (Y - Y_pred[:, 0]) * np.all(T == 0, axis=1) / propensities[:, 0]
+                for t in np.arange(T.shape[1]):
+                    T_counter = np.zeros(T.shape)
+                    T_counter[:, t] = 1
+                    Y_pred[:, t + 1] = self._model_regression.predict(np.hstack([XW, T_counter]))
+                    Y_pred[:, t + 1] += (Y - Y_pred[:, t + 1]) * (T[:, t] == 1) / propensities[:, t + 1]
+                return Y_pred
+
+        class ModelFinal:
+            def __init__(self, model_final, featurizer, multitask_model_final):
+                self._model_final = clone(model_final, safe=False)
+                self._featurizer = clone(featurizer, safe=False)
+                self._multitask_model_final = multitask_model_final
+                return
+
+            def _filter_none_kwargs(self, **kwargs):
+                out_kwargs = {}
+                for key, value in kwargs.items():
+                    if value is not None:
+                        out_kwargs[key] = value
+                return out_kwargs
+
+            def fit(self, Y, T, X=None, W=None, nuisances=None, *, sample_weight=None, sample_var=None):
+                Y_pred, = nuisances
+                if (X is not None) and (self._featurizer is not None):
+                    X = self._featurizer.fit_transform(X)
+                filtered_kwargs = _filter_none_kwargs(sample_weight=sample_weight, sample_var=sample_var)
+                if self._multitask_model_final:
+                    self.model_cate = clone(self._model_final, safe=False).fit(
+                        X, Y_pred[:, 1:] - Y_pred[:, [0]], **filtered_kwargs)
+                else:
+                    self.models_cate = [clone(self._model_final, safe=False).fit(X, Y_pred[:, t] - Y_pred[:, 0],
+                                                                                 **filtered_kwargs)
+                                        for t in np.arange(1, Y_pred.shape[1])]
+                return self
+
+            def predict(self, X=None):
+                if (X is not None) and (self._featurizer is not None):
+                    X = self._featurizer.transform(X)
+                if self._multitask_model_final:
+                    return self.model_cate.predict(X)
+                else:
+                    return np.array([mdl.predict(X) for mdl in self.models_cate]).T
+
+            def score(self, Y, T, X=None, W=None, nuisances=None, *, sample_weight=None, sample_var=None):

```

Comment:
Again, consider making nuisances keyword-only

```suggestion
            def score(self, Y, T, X=None, W=None, *, nuisances, sample_weight=None, sample_var=None):
```